PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs

```diff
@@ -132,6 +144,124 @@ public void SetRunspace(bool overrideExecutionPolicy)
         }
 
         /// <summary>
+        /// Attaches the HostService to a local runspace already in execution
+        /// </summary>
+        public void AttachToRunspace(uint pid)
+        {
+            if (_callback == null)
+            {
+                _callback = OperationContext.Current.GetCallbackChannel<IDebugEngineCallback>();
+            }
+
+            // Attaching leverages cmdlets introduced in PSv5
+            if (_installedPowerShellVersion < RequiredPowerShellVersionForProcessAttach)
+            {
+                MessageBox.Show(Constants.ProcessAttachVersionError, "Invalid Version", MessageBoxButton.OK, MessageBoxImage.Warning);
+                return;
+            }
+
+            // Enter into to-attach process which will swap out the current runspace
+            _attaching = true;
+            _needToOpen = true;
+            _currentPowerShell = PowerShell.Create();
+            _currentPowerShell.Runspace = _runspace;
+            _currentPowerShell.AddCommand("Enter-PSHostProcess").AddParameter("Id", pid.ToString());
+            _currentPowerShell.Invoke();
+
+            // wait for invoke to finish swapping the runspaces
+            _attachRequestEvent.WaitOne(5000);
+
+            // rehook up the event handling
+            _runspace.Debugger.DebuggerStop += Debugger_DebuggerStop;
+            _runspace.Debugger.BreakpointUpdated += Debugger_BreakpointUpdated;
+            _runspace.StateChanged += _runspace_StateChanged;
+            _runspace.AvailabilityChanged += _runspace_AvailabilityChanged;
+
+            // debug the runspace
+            _currentPowerShell.Runspace = _runspace;
+            _currentPowerShell.Commands.Clear();
+            _currentPowerShell.AddCommand("Debug-Runspace").AddParameter("Id", "1");

```

Comment:
Andre and I talked about it this morning. Consensus we reached is similar to what you're saying. Ping PowerShell contacts (Paul) to affirm this assumption. For now go with it as it seems to be the case. But add an issue for later to make a dialog if there are multiple runspaces to choose from so the user can specify.